### PR TITLE
remove jquerycssmenu.css, replace ul/li by button tag

### PR DESCRIPTION
**Pull request by tenzap**
This is because jquerycssmenu license doesn't permit redistribution, and so is problematic for Debian inclusion.

We replace the code by custom CSS using <button> tag instead of the ul/li
**Visual output is unchanged.**

License of jquerycssmenu.css is actually unsure and not to be found. In the past I attributed it to this license without being sure:

```
License: Dynamic-Drive-CSS-Library
 Dynamic Drive CSS Library Usage Terms <http://www.dynamicdrive.com/style/>
 .
   The CSS codes offered in [37]Dynamic Drive CSS Library are free for use
   on both personal and commercial sites. However, the following usage
   terms exist, and must be understood/ abided by first:
    1. Users may use any CSS code offered for download on [38]Dynamic
       Drive's CSS Library, free of charge, on both personal and
       commercial web sites. This includes web designers who wish to use
       our CSS codes in their paid web site projects.
    2. Users may NOT, however, redistribute or repost/ resell for download
       any CSS code (including any graphics) found in the CSS library.
       Redistribution is defined as re-offering our codes for download in
       any fashion, whether on a competing web site, an application that
       generates code snippets, or a CD-ROM collection of CSS/JavaScript
       codes etc. Some examples of what is acceptable and what is not are:
          + Acceptable:
            -Use our CSS codes on any personal or commercial web site to
            aid in its appearance/ usability.
            -As a web designer, use our CSS codes in your paid projects
            for your client web sites.
            -As a software developer, use our CSS codes within a
            application/ program as part of its interface, such as a CSS
            menu being used as the program's navigation interface. The
            program itself can be distributable.
          + Not Acceptable:
            -Put our CSS codes (including graphics) on another webmaster
            type site for others to download.
            -Use our CSS codes in any type of service or application
            whereby our codes are part of the product offerings
            themselves.
            -Put our CSS code in any other types of medium for
            redistribution, such as a CD-ROM that consists of, but not
            limited to, webmaster codes and web graphics.
    3. Users agree not to remove/ edit the credit notice within the CSS
       source code, or claim the code to be work of their own.
    4. Users are not required to link back to Dynamic Drive CSS library to
       use our CSS codes, as much as they are appreciated. :)
    5. Users agree not to hold Dynamic Drive liable for any damages
       resulted from proper or improper use of any of the CSS codes found
       in the CSS Library. Use at your own risk.
 .
 By using any code offered in our CSS Library, it is understood you have read
 and agreed to the above usage terms.
```

### CODE REVIEW

File: application/views/main/messages/navigation.php [hunk 1]
```diff
@@ -33,55 +32,69 @@
...
+	<div class="bttn-group">
+		<button><a href="javascript:void(0);" class="select_all_button button"><?php echo tr('Select all');?></a></button>
+		<button><a href="javascript:void(0);" class="clear_all_button button"><?php echo tr('Deselect all');?></a></button>
+	</div>
+
+	<div class="bttn-group">
 		<?php if ($this->uri->segment(2) === 'conversation' && $this->uri->segment(4) === 'inbox') :
 			if ($this->uri->segment(6) !== '6') : ?>
-		<li><a href="javascript:void(0);" class="spam_button button"><?php echo tr('Report spam');?></a></li>
+		<button><a href="javascript:void(0);" class="spam_button button"><?php echo tr('Report spam');?></a></button>
```
Comment (kingster) [inline application/views/main/messages/navigation.php:53]: Is it possible to not change every place to button? The reason I ask is I couldn't validate if having a tag inside a button is the right way or not

https://www.w3docs.com/snippets/html/how-to-create-an-html-button-that-acts-like-a-link.html

If we can continue using li and a with the new css classes that would be great. Spans could be another option.

**Comment by tenzap** [comment]
@kingster is this one ok?

**Comment by tenzap** [comment]
@kingster, could you please approve this one? It is the change that is in the Debian package, and permits to conform to DFSG.

**Review by kingster — APPROVED** [review]

**Comment by kingster** [comment]
👍, lets go ahead

**Pull request merged**